test/linalg.jl

```diff
@@ -5,69 +5,101 @@
+        @test @inferred(one(one(SMatrix{2,2,Int}))) === @SMatrix [1 0; 0 1]
+
+        @test @inferred(one(MMatrix{2,2,Int}))::MMatrix == @MMatrix [1 0; 0 1]
+        @test @inferred(one(MMatrix{2,2}))::MMatrix == @MMatrix [1.0 0.0; 0.0 1.0]
+        @test @inferred(one(MMatrix{2}))::MMatrix == @MMatrix [1.0 0.0; 0.0 1.0]
     end
 
     @testset "eye()" begin
-        @test eye(SMatrix{2,2,Int}) === @SMatrix [1 0; 0 1]
-        @test eye(SMatrix{2,2}) === @SMatrix [1.0 0.0; 0.0 1.0]
-        @test eye(SMatrix{2}) === @SMatrix [1.0 0.0; 0.0 1.0]
+        @test @inferred(eye(SMatrix{2,2,Int})) === @SMatrix [1 0; 0 1]
+        @test @inferred(eye(SMatrix{2,2})) === @SMatrix [1.0 0.0; 0.0 1.0]
+        @test @inferred(eye(SMatrix{2})) === @SMatrix [1.0 0.0; 0.0 1.0]
+        @test @inferred(eye(eye(SMatrix{2,2,Int}))) === @SMatrix [1 0; 0 1]
 
-        @test eye(SMatrix{2,3,Int}) === @SMatrix [1 0 0; 0 1 0]
-        @test eye(SMatrix{2,3}) === @SMatrix [1.0 0.0 0.0; 0.0 1.0 0.0]
+        @test @inferred(eye(SMatrix{2,3,Int})) === @SMatrix [1 0 0; 0 1 0]
+        @test @inferred(eye(SMatrix{2,3})) === @SMatrix [1.0 0.0 0.0; 0.0 1.0 0.0]
 
-        @test eye(MMatrix{2,2,Int})::MMatrix == @MMatrix [1 0; 0 1]
-        @test eye(MMatrix{2,2})::MMatrix == @MMatrix [1.0 0.0; 0.0 1.0]
-        @test eye(MMatrix{2})::MMatrix == @MMatrix [1.0 0.0; 0.0 1.0]
+        @test @inferred(eye(MMatrix{2,2,Int}))::MMatrix == @MMatrix [1 0; 0 1]
+        @test @inferred(eye(MMatrix{2,2}))::MMatrix == @MMatrix [1.0 0.0; 0.0 1.0]
+        @test @inferred(eye(MMatrix{2}))::MMatrix == @MMatrix [1.0 0.0; 0.0 1.0]
     end
 
     @testset "cross()" begin
-        @test cross(SVector(1,2,3), SVector(4,5,6)) === SVector(-3, 6, -3)
+        @test @inferred(cross(SVector(1,2,3), SVector(4,5,6))) === SVector(-3, 6, -3)
+        @test @inferred(cross(SVector(1,2), SVector(4,5))) === -3
+
     end
 
     @testset "transpose() and conj()" begin
-        @test conj(SVector(1+im, 2+im)) === SVector(1-im, 2-im)
+        @test @inferred(conj(SVector(1+im, 2+im))) === SVector(1-im, 2-im)
 
-        @test @SVector([1, 2, 3]).' === @SMatrix([1 2 3])
-        @test @SMatrix([1 2; 0 3]).' === @SMatrix([1 0; 2 3])
-        @test @SMatrix([1 2 3; 4 5 6]).' === @SMatrix([1 4; 2 5; 3 6])
+        @test @inferred(transpose(@SVector([1, 2, 3]))) === @SMatrix([1 2 3])
+        @test @inferred(transpose(@SMatrix([1 2; 0 3]))) === @SMatrix([1 0; 2 3])
+        @test @inferred(transpose(@SMatrix([1 2 3; 4 5 6]))) === @SMatrix([1 4; 2 5; 3 6])
 
-        @test @SVector([1, 2, 3])' === @SMatrix([1 2 3])
-        @test @SMatrix([1 2; 0 3])' === @SMatrix([1 0; 2 3])
-        @test @SMatrix([1 2 3; 4 5 6])' === @SMatrix([1 4; 2 5; 3 6])
+        @test @inferred(ctranspose(@SVector([1, 2, 3]))) === @SMatrix([1 2 3])
+        @test @inferred(ctranspose(@SMatrix([1 2; 0 3]))) === @SMatrix([1 0; 2 3])
+        @test @inferred(ctranspose(@SMatrix([1 2 3; 4 5 6]))) === @SMatrix([1 4; 2 5; 3 6])
```

Comment:
That's a lot of `@inferred`.  You need a custom `@testinferred` macro or some kind of helper to make these easier to look at ;-)